test/fit_tests/stream-monitor/stream_sources/log_type.py

```diff
@@ -0,0 +1,88 @@
+from logging import Logger, INFO, getLoggerClass
+from logging import _srcfile  # yes, this is evil. No, we don't have a choice.
+from .monitor_abc import StreamMonitorABC
+import sys
+
+class LoggingMarker(StreamMonitorABC):
+    _all_blocks = 0
+    def __init__(self):
+        # import HERE to prevent taking over logfiles for things
+        # like 'nosetests --help'.
+        import flogging
+        self.__print_to = None
+        # Uncomment next line to view steps to console live
+        # self.__print_to = sys.stderr
+        self.__loggers = self.__find_loggers()
+        self.__test_cnt = 0
+
+    @classmethod
+    def enabled_for_nose(self):
+        return True
+
+    def handle_begin(self):
+        self.__loggers = self.__find_loggers()
+        self.__mark_all_loggers(str(self._all_blocks), 'start-of-all-tests')
+        self._all_blocks += 1
+
+    def handle_start_test(self, test):
+        ts = '+{0}'.format(self.__test_cnt)
+        self.__mark_all_loggers(ts, '-------STARTING TEST: %s--------', str(test))
+
+    def handle_stop_test(self, test):
+        ts = '-{0}'.format(self.__test_cnt)
+        self.__test_cnt += 1
+        self.__mark_all_loggers(ts, '--------ENDING TEST: %s--------', str(test))
+
+    def __mark_all_in_logger(self, logger, msg, args, exc_info=None, extra=None):
+        handlers = getattr(logger, 'handlers', [])
+        if len(handlers) == 0:
+            return  # Nothing to do!
+        if _srcfile:
+            #IronPython doesn't track Python frames, so findCaller raises an
+            #exception on some versions of IronPython. We trap it here so that
+            #IronPython can use logging.
+            try:
+                fn, lno, func = logger.findCaller()
+            except ValueError:
+                fn, lno, func = "(unknown file)", 0, "(unknown function)"
+        else:
+            fn, lno, func = "(unknown file)", 0, "(unknown function)"
+        if exc_info:
+            if not isinstance(exc_info, tuple):
+                exc_info = sys.exc_info()
+        record = logger.makeRecord(logger.name, INFO, fn, lno, msg, args, exc_info, func, extra)
+        for handler in handlers:
+            handler.handle(record)
+
+    def __mark_all_loggers(self, bracket, fmat, *args, **kwargs):
+        rep_cnt = int(70/len(bracket))
```

Comment:
I'm finding in a full log, the rep_cnt of 70 make it hard to pick out the test name, can you drop it to 8 so it stands out a bit more. 
```
2016-12-08 16:13:16,327 547 MainProcess /emc/hohene/sandbox-github/rackhd/stu-log/test/fit_tests/stream-monitor/stream_sources/log_type.py:handle_start_test@29 greenlet:main INFO +0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+\
0+0+0+0+0+0
2016-12-08 16:13:16,327 547 MainProcess /emc/hohene/sandbox-github/rackhd/stu-log/test/fit_tests/stream-monitor/stream_sources/log_type.py:handle_start_test@29 greenlet:main INFO -------STARTING TEST: test_api_11_catalogs (rackhd11.test_r\
ackhd11_api_catalogs.rackhd11_api_catalogs)--------
2016-12-08 16:13:16,327 547 MainProcess /emc/hohene/sandbox-github/rackhd/stu-log/test/fit_tests/stream-monitor/stream_sources/log_type.py:handle_start_test@29 greenlet:main INFO +0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+0+\
0+0+0+0+0+0
```
```
2016-12-08 16:13:16,327 547 MainProcess /emc/hohene/sandbox-github/rackhd/stu-log/test/fit_tests/stream-monitor/stream_sources/log_type.py:handle_start_test@29 greenlet:main INFO +0+0+0+0
2016-12-08 16:13:16,327 547 MainProcess /emc/hohene/sandbox-github/rackhd/stu-log/test/fit_tests/stream-monitor/stream_sources/log_type.py:handle_start_test@29 greenlet:main INFO -------STARTING TEST: test_api_11_catalogs (rackhd11.test_r\
ackhd11_api_catalogs.rackhd11_api_catalogs)--------
2016-12-08 16:13:16,327 547 MainProcess /emc/hohene/sandbox-github/rackhd/stu-log/test/fit_tests/stream-monitor/stream_sources/log_type.py:handle_start_test@29 greenlet:main INFO +0+0+0+0
```